### ISSUE TITLE
Clarifying Trust of Materials Property Predictions Using Neural Networks with Distribution-Specific Uncertainty Quantification

### ISSUE THREAD

**Issue by shnakazawa** [TeamAidemy/ds-paper-summaries#22]
Gruich, Cameron, Varun Madhavan, Yixin Wang, and Bryan Goldsmith. 2023. “Clarifying Trust of Materials Property Predictions Using Neural Networks with Distribution-Specific Uncertainty Quantification.” arXiv [cs.LG]. arXiv. http://arxiv.org/abs/2302.02595.

- 機械学習を用いた材料開発の効率化には不確実性の適切な評価が重要。
- Open Catalyst 2020の触媒データを対象に、ニューラルネットワークによる触媒性質の予測モデルを構築。そして予測の不確実性の定量手法3種を比較。
- Evidential regressionが最適と結論。さらにRecalibrationを行うことで不確実性の信頼性が向上。
- 用いられた評価方法は材料科学に限らず機械学習による予測の信頼性評価に広く応用可能。

雑にまとめると「"k-fold分割してAccuracy見て評価" よりも良いモデルの評価方法がある」というお話。実務のみならずコンペでも生きそう。
「モデルが良い精度のアウトプットを返す」に加え、「その精度がどの程度信頼できるのか」という指標も得られる点が紹介されている手法群の強み。

## Abstract
> It is critical that machine learning (ML) model predictions be trustworthy for high-throughput catalyst discovery approaches. Uncertainty quantification (UQ) methods allow estimation of the trustworthiness of an ML model, but these methods have not been well explored in the field of heterogeneous catalysis. Herein, we investigate different UQ methods applied to a crystal graph convolutional neural network (CGCNN) to predict adsorption energies of molecules on alloys from the Open Catalyst 2020 (OC20) dataset, the largest existing heterogeneous catalyst dataset. We apply three UQ methods to the adsorption energy predictions, namely k-fold ensembling, Monte Carlo dropout, and evidential regression. The effectiveness of each UQ method is assessed based on accuracy, sharpness, dispersion, calibration, and tightness. Evidential regression is demonstrated to be a powerful approach for rapidly obtaining tunable, competitively trustworthy UQ estimates for heterogeneous catalysis applications when using neural networks. Recalibration of model uncertainties is shown to be essential in practical screening applications of catalysts using uncertainties.

(DeepL翻訳)

ハイスループットな触媒探索を行うためには、機械学習（ML）モデルの予測値が信頼できるものであることが重要である。不確実性定量化（UQ）法は、MLモデルの信頼性を推定することができるが、不均一系触媒の分野ではまだ十分に検討されていない。ここでは、既存の最大規模の不均一系触媒データセットであるOpen Catalyst 2020 (OC20) データセットから、合金への分子の吸着エネルギーを予測するために、結晶グラフ畳み込みニューラルネットワーク (CGCNN) に適用するさまざまなUQ手法を検討する。吸着エネルギー予測には、k-foldアンサンブル、モンテカルロドロップアウト、エビデンス回帰という3つのUQ手法を適用した。各UQ法の有効性は、精度、シャープネス、分散、キャリブレーション、タイトネスに基づいて評価される。エビデンシャル回帰は、ニューラルネットワークを使用した場合、不均一系触媒のアプリケーションにおいて、調整可能で競争上信頼できるUQ推定値を迅速に得るための強力なアプローチであることが実証された。不確実性を利用した触媒の実用的なスクリーニングアプリケーションでは、モデルの不確実性の再キャリブレーションが不可欠であることが示された。

## データセット

- [Open Catalyst 2020](https://github.com/Open-Catalyst-Project/ocp/blob/main/DATASET.md)

## 解決した課題/先行研究との比較

- 機械学習を用いた材料開発の研究・応用が進んでいる。本論文では触媒の開発にフォーカス。
- 触媒性質予測モデルの開発が進み、予測の高精度化は進んできた。一方で、予測結果がどれほど信頼できるものかは明確にしにくかったり、計算量的に現実的でなかったりといった課題があった。

## 技術・手法のポイント

- 本論文ではOpen Catalyst 2020データセットを対象に、以下の3つの手法を比較した。具体的な方法については引用文献参照。
	- k-fold ensembling (Cross-validationのようにk分割→k個のモデルの出力をアンサンブル)
	- Monte Carlo dropout 
	- Evidential regression
- いずれの手法も予測値が複数出力されるため、それらの平均値μと標準偏差σを求めることができる。
- ニューラルネットワーク (Crystal Graph Convolutional Neural Network: CGCNN) をトレーニング→入力データに対する予測値と、その平均値、標準偏差を出力→予測の不確実性を定量評価。

[![Image from Gyazo](https://i.gyazo.com/e9e7a6dd5c478a2c198dac0ea8521deb.png)](https://gyazo.com/e9e7a6dd5c478a2c198dac0ea8521deb)

## 評価指標

以下の項目で不確実性を定量。図はその結果を示す。

- **Accuracy**
	- 予測値が実測値とどの程度ずれているかの指標。具体的には、MAE, RMSE, MDAE, MARPD, R^2, Rのこと。
[![Image from Gyazo](https://i.gyazo.com/d84134ed0ed1fbc70059fb792ec70539.png)](https://gyazo.com/d84134ed0ed1fbc70059fb792ec70539)
	- Evidential Regressionが最も良い

- **Sharpness**/**Dispersion**
	- Sharpness: 論文中の数式(7)。分散の平均の平方根。
 	- Dispersion: IQR (第3四部位数 - 第1四部位数), 変動係数 (Coefficient of variation: 標準偏差/平均)。
	- いずれも予測値にどの程度幅があるかの指標。数字が大きい = 予測結果に幅がある = 人が選ぶ選択肢が増える = 良いこと、と主張。
		- （サマリー作者談：この主張が適切かは自身の目的によるだろう。Materials Informaticsにおいては結果の多様性が重要となるケースも多い。）
[![Image from Gyazo](https://i.gyazo.com/bdd0f0875040e9fc2e2c47e7d57288d9.png)](https://gyazo.com/bdd0f0875040e9fc2e2c47e7d57288d9)
	- Evidential Regressionが最も良い

- **Calibration**
	- 予測結果の発生確率分布と、実際の発生確率分布が、どの程度近いか？
		- Accuracyは"値"の比較、Calibrationは"分布"の比較。
	- Reliability diagram plot: 実測値と予測値の平均値の差（をノーマライズしたもの）をプロット。
		- 対角線に沿っている = 予測値と実測値がった点で表されるモデルがよりcalibrateされているといえる。
	- "Re-calibration"とは、不確実性の推定を改善するための後処理。
		- scikit-learnに入っているBrent’s methodを使用。
		- > The constant was chosen via a black-box optimization algorithm

	- Reliability diagram plot (ハイパーパラメータによる差)
		- (**a**) デフォルトではMC-Dropout がベスト。(**b, c**) パラメータ調整によりEvidential Regressionがベストに。
[![Image from Gyazo](https://i.gyazo.com/af807a37304376d82722636181015260.png)](https://gyazo.com/af807a37304376d82722636181015260)

	- Reliability diagram plot (Re-calibration前後の比較)
		- Recalibrationはいずれの手法にも有効にはたらき、ミスキャリブレーションの問題は小さくなる。
[![Image from Gyazo](https://i.gyazo.com/023d5bd91088e633d34167a06920ed3b.png)](https://gyazo.com/023d5bd91088e633d34167a06920ed3b)

- **Tightness**
	- 予測値の幅が必要十分か。
 	- 不確実区間 (予測値の平均μと標準偏差σから、予測値が正規分布に従うと仮定し導出) を1~99%まで1%刻みで数値化し、「それぞれの幅」と「実測値との距離」の比の対数値をとり、99個の数字の平均をとり、正負逆転させたもの。数値が小さいほど良い。
[![Image from Gyazo](https://i.gyazo.com/a5cdf18307b44ee18304d93694946ac4.png)](https://gyazo.com/a5cdf18307b44ee18304d93694946ac4)
	- Evidential Regressionが最も良い。

ほぼすべての指標で、Evidential regression が最も高精度に不確実性を推定できていた。

## 残された課題・議論

- トレーニングデータがテストデータを代表する (トレーニングデータとテストデータの性質が同じ)と仮定しているため、常に適切な推定ができるとは限らないと指摘している。
- 小規模なデータセットの場合、信頼性の低い予測が得られる可能性があるとも議論。
	- すなわち、一般的な機械学習モデル、Cross-validationに対して言われる話と同じ。

## 重要な引用

- ニューラルネットワークを用いた予測における不確実性の定量に関する総説
	- Abdar, Moloud, Farhad Pourpanah, Sadiq Hussain, Dana Rezazadegan, Li Liu, Mohammad Ghavamzadeh, Paul Fieguth, et al. 	2020. “A Review of Uncertainty Quantification in Deep Learning: Techniques, Applications and Challenges.” arXiv [cs.LG]. arXiv. http://arxiv.org/abs/2011.06225.
- MIにおけるニューラルネットワーク予測の不確実性定量手法の先行研究
	- Tran, Kevin, Willie Neiswanger, Junwoong Yoon, Qingyang Zhang, Eric Xing, and Zachary W. Ulissi. 2019. “Methods for Comparing Uncertainty Quantifications for Material Property Predictions.” arXiv [cond-Mat.mtrl-Sci]. arXiv. http://arxiv.org/abs/1912.10066.
	- Hu, Yuge, Joseph Musielewicz, Zachary Ulissi, and Andrew J. Medford. 2022. “Robust and Scalable Uncertainty Estimation with Conformal Prediction for Machine-Learned Interatomic Potentials.” arXiv [physics.chem-Ph]. arXiv. http://arxiv.org/abs/2208.08337.
- k-fold  ensemblingについて
	- Dietterich, Thomas G. 2000. “Ensemble Methods in Machine Learning.” In Multiple Classifier Systems, 1–15. Springer Berlin Heidelberg.
	- Lakshminarayanan, Balaji, Alexander Pritzel, and Charles Blundell. 2017. “Simple and Scalable Predictive Uncertainty Estimation Using Deep Ensembles.” Advances in neural information processing systems.
	- Liu, Jeremiah Zhe, John Paisley, Marianthi-Anna Kioumourtzoglou, and Brent Coull. 2019. “Accurate Uncertainty Estimation and Decomposition in Ensemble Learning.” Advances in neural information processing systems.
- Monte Carlo dropoutについて
	- Gal, Yarin, and Zoubin Ghahramani. 2015. “Dropout as a Bayesian Approximation: Representing Model Uncertainty in Deep Learning.” arXiv [stat.ML]. arXiv. http://arxiv.org/abs/1506.02142.
- Evidential regressionについて
	- Amini, Alexander, Wilko Schwarting, Ava Soleimany, and Daniela Rus. 2020. “Deep Evidential Regression.” Advances in neural information processing systems.
	- Soleimany, Ava P., Alexander Amini, Samuel Goldman, Daniela Rus, Sangeeta N. Bhatia, and Connor W. Coley. 2021. “Evidential Deep Learning for Guided Molecular Property Prediction and Discovery.” ACS Central Science 7 (8): 1356–67.
- Calibrationについて
	- Gneiting, Tilmann, Fadoua Balabdaoui, and Adrian E. Raftery. 2007. “Probabilistic Forecasts, Calibration and Sharpness.” Journal of the Royal Statistical Society. Series B, Statistical Methodology 69 (2): 243–68.
	- Gneiting, Tilmann, and Matthias Katzfuss. 2014. “Probabilistic Forecasting.” Annual Review of Statistics and Its Application 1 (1): 125–51.
	- Guo, Chuan, Geoff Pleiss, Yu Sun, and Kilian Q. Weinberger. 06--11 Aug 2017. “On Calibration of Modern Neural Networks.” In Proceedings of the 34th International Conference on Machine Learning, edited by Doina Precup and Yee Whye Teh, 70:1321–30. Proceedings of Machine Learning Research. PMLR.
	- Levi, Dan, Liran Gispan, Niv Giladi, and Ethan Fetaya. 2022. “Evaluating and Calibrating Uncertainty Prediction in Regression Tasks.” Sensors  22 (15). https://doi.org/10.3390/s22155540.
	- Pernot, Pascal. 2022. “The Long Road to Calibrated Prediction Uncertainty in Computational Chemistry.” The Journal of Chemical Physics 156 (11): 114109.
- Recalibrationについて
	- Zhao, Shengjia, Tengyu Ma, and Stefano Ermon. 13--18 Jul 2020. “Individual Calibration with Randomized Forecasting.” In Proceedings of the 37th International Conference on Machine Learning, edited by Hal Daumé Iii and Aarti Singh, 119:11387–97. Proceedings of Machine Learning Research. PMLR.
	- Pedregosa, Fabian. 2011. “Scikit-Learn: Machine Learning in Python.” Journal of Machine Learning Research: JMLR 12: 2825–30.
- Tightnessについて
	- Gneiting, Tilmann, and Adrian E. Raftery. 2007. “Strictly Proper Scoring Rules, Prediction, and Estimation.” Journal of the American Statistical Association 102 (477): 359–78.
